### ISSUE TITLE
[BUG] UnhandledPromiseRejectionWarning caused by constructor class 

### ISSUE THREAD

**Issue by henriquecustodia** [compodoc/compodoc#105]
##### **Overview of the issue**

I'm trying to generate the documentation but I get a error in a file.

command:
```
./node_modules/.bin/compodoc -p src/tsconfig.json
``` 

Error: 
```
(node:25557) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): RangeError: Invalid array length
```

I think this problem is the constructor in CustomXSRFStrategy
```
import { CookieXSRFStrategy } from '@angular/http';

export class CustomXSRFStrategy extends CookieXSRFStrategy {
    constructor() {
        super('csrftoken', 'X-CSRFToken');
    }
}
```
When I remove the constructor class no errors happens

Complete log:
```
[13:03:48] Using tsconfig : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/tsconfig.json
[13:03:48] Including      : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/app/app.module.ts
[13:03:48] Including      : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/app/core/classes/xrfs-strategy/xrfs-strategy.ts
[13:03:48] Including      : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/app/core/core.module.ts
[13:03:48] Including      : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/app/core/interfaces/service.ts
[13:03:48] Including      : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/app/core/services/http-interceptor/http-interceptor.factory.ts
[13:03:48] Including      : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/app/core/services/http-interceptor/http-interceptor.module.ts
[13:03:48] Including      : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/app/core/services/services.module.ts
[13:03:48] Ignoring       : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/app/features/categories/categories.component.spec.ts
[13:03:48] Including      : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/app/features/categories/categories.component.ts
[13:03:48] Including      : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/app/features/categories/categories.module.ts
[13:03:48] Ignoring       : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/app/features/checkout/checkout.component.spec.ts
[13:03:48] Including      : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/app/features/checkout/checkout.component.ts
[13:03:48] Including      : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/app/features/checkout/checkout.module.ts
[13:03:48] Including      : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/app/features/products/product.model.ts
[13:03:48] Ignoring       : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/app/features/products/products.component.spec.ts
[13:03:48] Including      : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/app/features/products/products.component.ts
[13:03:48] Including      : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/app/features/products/products.module.ts
[13:03:48] Including      : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/app/features/products/products.serializer.ts
[13:03:48] Ignoring       : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/app/features/products/products.service.spec.ts
[13:03:48] Including      : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/app/features/products/products.service.ts
[13:03:48] Ignoring       : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/app/features/profile/profile.component.spec.ts
[13:03:48] Including      : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/app/features/profile/profile.component.ts
[13:03:48] Including      : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/app/features/profile/profile.module.ts
[13:03:48] Including      : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/app/shared/components/components.module.ts
[13:03:48] Ignoring       : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/app/shared/components/footer/footer.component.spec.ts
[13:03:48] Including      : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/app/shared/components/footer/footer.component.ts
[13:03:48] Including      : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/app/shared/components/footer/footer.module.ts
[13:03:48] Ignoring       : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/app/shared/components/header/header.component.spec.ts
[13:03:48] Including      : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/app/shared/components/header/header.component.ts
[13:03:48] Including      : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/app/shared/components/header/header.module.ts
[13:03:48] Including      : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/app/shared/shared.module.ts
[13:03:48] Including      : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/environments/environment.prod.ts
[13:03:48] Including      : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/environments/environment.ts
[13:03:48] Including      : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/example/app-routing.module.ts
[13:03:48] Ignoring       : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/example/app.component.spec.ts
[13:03:48] Including      : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/example/app.component.ts
[13:03:48] Including      : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/example/app.module.ts
[13:03:48] Including      : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/main.ts
[13:03:48] Including      : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/polyfills.ts
[13:03:48] Including      : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/test.ts
[13:03:48] Searching package.json file
[13:03:48] package.json file found
[13:03:48] Searching README.md file
[13:03:48] README.md file found
[13:03:48] Get dependencies data
[13:03:48] parsing        : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/app/app.module.ts
[13:03:50] debug          : Ae2Module:
[13:03:50]                : - imports:
[13:03:50]                : 	- CoreModule.forRoot()
[13:03:50]                : 	- SharedModule
[13:03:50]                : - exports:
[13:03:50]                : 	- CoreModule
[13:03:50]                : 	- SharedModule
[13:03:50]                : - declarations:
[13:03:50]                : 	- ProfileComponent
[13:03:50] parsing        : /home/henriquecustodia/Workspace/Angular/angular-ecommerce2/src/app/core/classes/xrfs-strategy/xrfs-strategy.ts
(node:26196) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): RangeError: Invalid array length

```

##### **Operating System, Node.js, npm, compodoc version(s)**
OS: Ubuntu 16.04
Node: 7.4.0
Npm: 4.0.5

```
node_modules/.bin/compodoc -V
0.0.39
```
##### **Angular configuration, a `package.json` file in the root folder**

``` {
  "name": "angular-ecommerce2",
  "version": "0.0.0",
  "license": "MIT",
  "angular-cli": {},
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "test": "ng test",
    "pree2e": "webdriver-manager update --standalone false --gecko false",
    "e2e": "protractor",
    "doc": "./node_modules/.bin/compodoc -p src/tsconfig.json"
  },
  "private": true,
  "dependencies": {
    "@angular/common": "^2.3.1",
    "@angular/compiler": "^2.3.1",
    "@angular/core": "^2.3.1",
    "@angular/forms": "^2.3.1",
    "@angular/http": "^2.3.1",
    "@angular/platform-browser": "^2.3.1",
    "@angular/platform-browser-dynamic": "^2.3.1",
    "@angular/router": "^3.3.1",
    "core-js": "^2.4.1",
    "lodash.forin": "^4.4.0",
    "lodash.noop": "^3.0.1",
    "ng2-interceptors": "^1.3.0-0",
    "rxjs": "^5.0.1",
    "ts-helpers": "^1.1.1",
    "zone.js": "^0.7.2"
  },
  "devDependencies": {
    "@angular/cli": "1.0.0-beta.30",
    "@angular/compiler-cli": "^2.3.1",
    "@types/jasmine": "2.5.38",
    "@types/lodash": "^4.14.52",
    "@types/node": "^6.0.42",
    "codelyzer": "~2.0.0-beta.1",
    "compodoc": "^0.0.39",
    "jasmine-core": "2.5.2",
    "jasmine-spec-reporter": "2.5.0",
    "karma": "1.2.0",
    "karma-chrome-launcher": "^2.0.0",
    "karma-cli": "^1.0.1",
    "karma-jasmine": "^1.0.2",
    "karma-remap-istanbul": "^0.2.1",
    "protractor": "~4.0.13",
    "ts-node": "1.2.1",
    "tslint": "^4.3.0",
    "typescript": "~2.0.3"
  }
}
```
##### **Compodoc installed globally or locally ?**
locally

**Comment by vogloblinsky** [comment]
Cannot reproduce the bug with just this code pasted in a .ts file :

```
import { CookieXSRFStrategy } from '@angular/http';

export class CustomXSRFStrategy extends CookieXSRFStrategy {
    constructor() {
        super('csrftoken', 'X-CSRFToken');
    }
}
```
Can you search in __node_modules/compodoc/dist/index.cli.js__ the function __getJSDocs__, and add this before the return.
```console.log(tsany.version);```
and give me your version

**Comment by henriquecustodia** [comment]
Thanks for reply. 

The version is `2.2.0`

**Comment by vogloblinsky** [comment]
Ok thanks.
The API i use for JSDoctags is not available in TypeScript 2.2.0
I will publish a release with a fix for that tomorrow.
I am waiting feedbacks on TypeScript repository for this changes.

**Comment by henriquecustodia** [comment]
Great! Thanks man :smile:
